src/index.d.ts

```diff
@@ -0,0 +1,9 @@
+
+import { Middleware, Dispatch, Action } from 'redux'
+
+export type OnActionCallback = (action: Action, dispatch: Dispatch, getState: () => any) => any
+
+export default function createOnActionMiddleware(): {
+    middleware: Middleware
+    onAction: <TAction extends string>(action: TAction, callback: OnActionCallback) => void
+}
```

Comment:
Apply indentation of 2 spaces and and new line at the end of file (conform to project's .editorconfig)